Zagrios/bs-manager
# [feature] Automating the wiki via GitHub Actions

## Scope



This Pull Request introduces automation to the Wiki, to allow contributors to edit the Wiki through Markdown pages. [I previously did this automation for another project](https://github.com/imLinguin/comet/pull/24), and figured contributing it to this project as well to accommodate documentation efforts such as desired in #468.

Accepting contributions via PR (and have it automated into the Wiki after the PR merged) will prevent vandalism through the mandatory review before merge, while allowing anyone to contribute and ask for feedback.

This feature might be in particular useful for @GaetanGrd, as they are the assigned Documentation Lead.

## Implementation



A GitHub Actions workflow is created by using [Andrew-Chen-Wang/github-wiki-action@v4](https://github.com/marketplace/actions/github-wiki-action), with it focusing on any Markdown files located in the `docs/wiki/` directory. The pipeline will only trigger when any changes are made to any files in `docs/wiki/`, on the `master` branch. This does include additions.

**This feature does make the `docs/wiki/` pages the source of truth, overwriting any current article in the wiki.**

## Screenshots

| before | after |
| ------ | ----- |
|![image](https://github.com/user-attachments/assets/424596bf-09dc-4ad8-a367-d280edbcfad1)|![image](https://github.com/user-attachments/assets/dc5ca00e-5603-4f76-8991-c6c10cc469cb) |
| Manual wiki editing|![image](https://github.com/user-attachments/assets/6a9649c9-d2b7-48b4-9ce8-a76ec50e3460)|

## How to Test

- Fork my forked/pull request repository
- Create initial wiki page via the GUI
    - The GitHub Action requires a pre-existing Wiki. It can not create a Wiki on its own.
- Change anything in the `docs/wiki/` directory.
- You should see the present `docs/wiki/*.md` files represented in the test repository's wiki.

Test results can be found in my [forked repository](https://github.com/kevin-wijnen/bs-manager). The initial page was a test page, with the current wiki's pages added as a commit before it being published through the CI/CD.



## Emoji Guide

**For reviewers: Emojis can be added to comments to call out blocking versus non-blocking feedback.**

E.g: Praise, minor suggestions, or clarifying questions that don’t block merging the PR.

> 🟢 Nice refactor!

> 🟡 Why was the default value removed?

E.g: Blocking feedback must be addressed before merging.

> 🔴 This change will break something important

|              |                |                                     |
| ------------ | -------------- | ----------------------------------- |
| Blocking     | 🔴 ❌ 🚨       | RED                                 |
| Non-blocking | 🟡 💡 🤔 💭    | Yellow, thinking, etc               |
| Praise       | 🟢 💚 😍 👍 🙌 | Green, hearts, positive emojis, etc |
